dev/com.ibm.ws.security.wim.adapter.ldap_fat/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/LoginPropertyTest.java

```diff
@@ -0,0 +1,290 @@
+
+        ldap.setRealm("LDAPRealm");
+        ldap.setHost("localhost");
+        ldap.setPort(String.valueOf(ds.getLdapPort()));
+        ldap.setBaseDN(DN);
+        ldap.setBindDN(InMemoryLDAPServer.getBindDN());
+        ldap.setBindPassword(InMemoryLDAPServer.getBindPassword());
+        ldap.setLdapType("Custom");
+        ldap.setLdapCache(new LdapCache(new AttributesCache(false, 0, 0, "0s"), new SearchResultsCache(false, 0, 0, "0s")));
+        for (String prop : loginProps) {
+            ldap.addLoginProperty(new LoginProperty(prop));
+        }
+
+        if (addFilters) {
+            ldap.setCustomFilters(new LdapFilters("(&(cn=%v)(objectclass=inetorgperson))", null, "user:postalCode", null, null));
+        }
+        server.getLdapRegistries().add(ldap);
+        updateConfigDynamically(libertyServer, server);
+    }
+
+    /**
+     * Log in the with the user's uid. No loginProperties. No userFilter defined.
+     * This is using all the defaults, including loginProperty and the generated user filter.
+     * Uid should work. Cn should fail.
+     * Additionally, we expect the user display name to be uid.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testDefaultConfig() throws Exception {
+        updateLibertyServer(false);
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userUid, "password"));
+        assertNull("Authentication should not succeed.", servlet.checkPassword(userCn, "password"));
+        assertEquals(userUid, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Log in the with the user's cn from userFilter. No loginProperties.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testNoLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true);
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userCn, "password"));
+    }
+
+    /**
+     * Log in the with the user's sn from loginProperties. Include userFilter.
+     * This test should issue a warning that the userFilter is ignored and
+     * the dynamically built filter is used.
+     * Additionally, we expect the user display name to be sn because loginProperty
+     * takse precedence.
+     *
+     * @throws Exception If there was an unexpected exception.
+     */
+    @Test
+    public void testLoginPropertyWithFilter() throws Exception {
+        updateLibertyServer(true, "sn");
+        assertDNsEqual("Authentication should succeed.",
+                       USER_DN, servlet.checkPassword(userSn, "password"));
+        assertNotNull("Should find CWIML4505W message.", libertyServer.waitForStringInLog("CWIML4505W"));
+        assertEquals(userSn, servlet.getUserDisplayName(USER_DN));
+    }
+
+    /**
+     * Log in the with the user's sn from loginProperties.
```

Comment:
I would probably just make this the same as the one above (use CN), and the key difference is NO userFilter was defined. It should perform identical to the test above I believe.